M5/code-reviews/widgetsController.js

```diff
@@ -0,0 +1,68 @@
+var express = require("express");
+var widgets = express.Router();
+var widgetsArray = require("../models/widget");
+
+/* code tested fine in Postman, but won't pass test POST, commented out for now
+const validateData = (req, res, next) => {
+    const checkInputDataType = (type, objContainsSameType) => {
+        for (const key in objContainsSameType) {
+            if (typeof objContainsSameType[key] !== type)
+                return { passed: false, invalid: key };
+        }
+        return { passed: true };
+    }
```

Comment:
break `checkInputDataType` out into its own separate definition, and call it from `validateData`